probe-rs/src/probe/jlink/mod.rs

```diff
@@ -941,6 +945,55 @@ impl DAPAccess for JLink {
     }
 }
 
+impl SwoAccess for JLink {
+    fn enable_swo(&mut self, config: &SwoConfig) -> Result<(), ProbeRsError> {
+        let jlink = self.handle.get_mut().unwrap();
+        self.swo_baud = Some(config.baud());
+        jlink
+            .swo_start_uart(config.baud(), SWO_BUFFER_SIZE.into())
+            .map_err(|e| ProbeRsError::Probe(DebugProbeError::ArchitectureSpecific(Box::new(e))))?;
+        Ok(())
+    }
+
+    fn disable_swo(&mut self) -> Result<(), ProbeRsError> {
+        let jlink = self.handle.get_mut().unwrap();
+        self.swo_baud = None;
+        jlink
+            .swo_stop()
+            .map_err(|e| ProbeRsError::Probe(DebugProbeError::ArchitectureSpecific(Box::new(e))))?;
+        Ok(())
+    }
+
+    fn read_swo_timeout(&mut self, timeout: std::time::Duration) -> Result<Vec<u8>, ProbeRsError> {
+        const MULTIPLIER: u32 = 2;
+
+        let end = std::time::Instant::now() + timeout;
+        let mut buf = vec![0; SWO_BUFFER_SIZE.into()];
+
+        let bytes_per_sec = self.swo_baud.unwrap() / 8;
+        let buffers_per_sec = std::cmp::max(1, bytes_per_sec / (buf.len() as u32)) * MULTIPLIER;
+        let poll_interval =
+            std::time::Duration::from_micros(1_000_000 / u64::from(buffers_per_sec));
```

Comment:
It would make sense to expose this `poll_interval` calculation as `SwoAccess::swo_poll_interval_hint()`, then other users can know how often to poll too.

Reply:
Here, the same applies as in my other comment. Let's put this idea into a tracking issue and start thinking about how to make this next step happen.

In particular because changes to `SwoAccess` affect all types of probes and not just the JLink one. So IMHO that should be part of a separate PR.

Reply:
`SwoAccess` already has the `swo_poll_interval_hint()` method, just if you don't implement it (e.g. here for jlink) it returns `None` by default, so since you already do the computation for it you might as well expose the method and use it internally too.

Reply:
Sorry, I didn't understand this is already in the code.

PR is updated by f501128 to take care of that.